lib/release_page.dart

```diff
@@ -47,57 +48,71 @@ class ReleasePage extends StatelessWidget {
       );
     }
 
-    return Scaffold(
-      appBar: AppBar(
-        title: Text('${device.name} ${device.version}'),
-        leading: Center(
-          child: IconButton(
-            style: IconButton.styleFrom(fixedSize: const Size(40, 40)),
-            onPressed: Navigator.of(context).pop,
-            icon: const Icon(YaruIcons.go_previous),
-          ),
+    if (model.state == DeviceState.needsReboot) {
+      WidgetsBinding.instance.addPostFrameCallback(
+        (_) => showConfirmationDialog(
+          context,
+          text: l10n.rebootConfirm,
+          okText: l10n.reboot,
+          onCancel: () => model.state = DeviceState.idle,
+          onConfirm: model.reboot,
         ),
-        elevation: 0,
-      ),
-      body: ListView(
-        padding: const EdgeInsets.symmetric(horizontal: 16),
-        children: releases
-            .map(
-              (release) => ReleaseCard(
-                release: release,
-                selected: release == selected,
-                onSelected: () =>
-                    context.read<DeviceModel>().selectedRelease = release,
+      );
+    }
+
+    return model.state == DeviceState.busy
+        ? const Center(child: YaruCircularProgressIndicator())
```

Comment:
Could we instead navigate back to the device page and perhaps show a spinner somewhere in the corner to indicate the busy status?

Reply:
Just to elaborate, I'm proposing this to avoid momentarily replacing the underlying content with a spinner that results in a "blink" effect. :)

Reply:
Yes that's a good idea. My intention was to prevent the user from trying to select and install another (or the same) release due to impatience. Do you think it's a good idea to display the loading indicator in the top right corner of the device page, either instead of the buttons, or next to the buttons which I'd disable?

Reply:
Or maybe we could leave it out of this PR because it's not needed for the reboot? We can discuss the busy status in the meeting later :)

Reply:
ok, good point - I'll just navigate back to the device page for now and display the reboot confirmation dialog there after the update finishes.